src/lib/topology/topology.py

```diff
@@ -0,0 +1,67 @@
+import asyncio
+from typing import List, Set, Dict, Iterable, Awaitable
+
+from lib.context import MetricsContext
+from lib.entities import entities_extractors
+from lib.entities.model import Entity
+from lib.metrics import GCPService
+
+
+async def fetch_topology(context: MetricsContext, project_id: str, services: List[GCPService], disabled_apis: Set[str]) \
+        -> Dict[GCPService, Iterable[Entity]]:
+
+    topology_tasks_by_service: Dict[GCPService, Awaitable[Iterable[Entity]]] = {}
+
+    for service in choose_services_for_topology_fetch(context, project_id, services, disabled_apis):
+        topology_function = entities_extractors[service.name].extractor(context, project_id, service)
+        topology_task = asyncio.create_task(topology_function)
+        topology_tasks_by_service[service] = topology_task
+
+    topology_by_service: Dict[GCPService, Iterable[Entity]] = {}
+
+    for service, task in topology_tasks_by_service.items():
+        topology_by_service[service] = await task
+
+    return topology_by_service
+
+
+def choose_services_for_topology_fetch(
```

Comment:
It's my duty to state my opinion: we are adding another iteration with more lines of code for a very simple task.
But again, it's subjective and I get the approach :), I approve anyway.